catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceQueryService.java

```diff
@@ -0,0 +1,386 @@
+
+    private static final String DEFAULT_CRON_STRING = "0 0 0 * * ?";
+
+    private static final String TRIGGER_NAME = "WorkspaceQueryTrigger";
+
+    private static WorkspaceQueryService instance = null;
+
+    private final QueryUpdateSubscriber queryUpdateSubscriber;
+
+    private final WorkspaceService workspaceService;
+
+    private final CatalogFramework catalogFramework;
+
+    private final FilterBuilder filterBuilder;
+
+    @SuppressWarnings("FieldCanBeLocal")
+    private Scheduler scheduler;
+
+    private SecurityService securityService;
+
+    private FilterService filterService;
+
+    private long queryTimeoutMinutes = DEFAULT_QUERY_TIMEOUT_MINUTES;
+
+    private JobDetail jobDetail;
+
+    /**
+     * @param queryUpdateSubscriber must be non-null
+     * @param workspaceService      must be non-null
+     * @param catalogFramework      must be non-null
+     * @param filterBuilder         must be non-null
+     * @param schedulerSupplier     must be non-null
+     * @param securityService       must be non-null
+     * @param filterService         must be non-null
+     */
+    public WorkspaceQueryService(QueryUpdateSubscriber queryUpdateSubscriber,
+            WorkspaceService workspaceService, CatalogFramework catalogFramework,
+            FilterBuilder filterBuilder, Supplier<Optional<Scheduler>> schedulerSupplier,
+            SecurityService securityService, FilterService filterService)
+            throws SchedulerException {
+
+        notNull(queryUpdateSubscriber, "queryUpdateSubscriber must be non-null");
+        notNull(workspaceService, "workspaceService must be non-null");
+        notNull(catalogFramework, "catalogFramework must be non-null");
+        notNull(filterBuilder, "filterBuilder must be non-null");
+        notNull(schedulerSupplier, "scheduleSupplier must be non-null");
+        notNull(securityService, "securityService must be non-null");
+        notNull(filterService, "filterService must be non-null");
+
+        this.queryUpdateSubscriber = queryUpdateSubscriber;
+        this.workspaceService = workspaceService;
+        this.catalogFramework = catalogFramework;
+        this.filterBuilder = filterBuilder;
+        this.securityService = securityService;
+        this.filterService = filterService;
+
+        Optional<Scheduler> schedulerOptional = schedulerSupplier.get();
+
+        instance = this;
+
+        if (schedulerOptional.isPresent()) {
+            scheduler = schedulerOptional.get();
+            jobDetail = newJob(QueryJob.class).withIdentity(JOB_IDENTITY)
+                    .build();
+            scheduler.scheduleJob(jobDetail,
+                    new CronString(DEFAULT_CRON_STRING, TRIGGER_NAME).get());
+            scheduler.start();
+        } else {
+            LOGGER.warn("unable to get a quartz scheduler object, email notifications will not run");
+        }
+
+    }
+
+    public static WorkspaceQueryService getInstance() {
+        return instance;
```

Comment:
I'm going to give this some thought. The quartz command needs to get a reference to the WorkspaceQueryService. But I think I get have the quartz command use FrameworkUtils to get the reference.